### PR TITLE
Gulpfile cleanup

### PR DESCRIPTION
**Pull request by NinjaManatee**
Simplified and organized the gulpfile which will make life easier for the project reorg

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -1,210 +1,182 @@
...
+const launcher = require("openfin-launcher");
+const path = require("path");
+const webpack = require("webpack-stream");
+
+// local
+const webpackFilesConfig = require("./build/webpack/webpack.files.js")
+const webpackServicesConfig = require("./build/webpack/webpack.services.js")
+// #endregion
+
+// #region Constants
```
Comment (carterbs) [inline gulpfile.js]: what's region do? or is that just for you?

File: gulpfile.js [hunk 2]
```diff
@@ -1,210 +1,182 @@
...
+	const source = [
+		path.join(srcPath, "components", "**", "*"),
+		"!" + path.join(srcPath, "components", "**", "*.jsx")];
+
+	// Don't copy files that we build
+	for (const key in componentsToBuild) {
+		source.push("!" + path.join(__dirname, componentsToBuild[key].entry));
+	}
+	
+	return merge(
```
Comment (carterbs) [inline gulpfile.js]: what's merge? does it merge the streams into one?

File: gulpfile.js [hunk 3]
```diff
@@ -1,210 +1,182 @@
...
-				return;
-			}
-			this.process.exit();
+	return launcher
+		.launchOpenFin({
+			configPath: startupConfig[env].serverConfig
+		})
+		.then(() => {
+			// OpenFin has closed so exit gulpfile
+			process.exit();
```
Comment (carterbs) [inline gulpfile.js]: wow. does this solve ctrl+c?????

File: gulpfile.js (hunk 1 above)
Comment (NinjaManatee) [reply gulpfile.js]: It is organizational. In VS code, it folds the region.

File: gulpfile.js (hunk 2 above)
Comment (NinjaManatee) [reply gulpfile.js]: Yes, so the task doesn't complete until all of the streams complete. https://www.npmjs.com/package/merge-stream

Comment (carterbs) [reply gulpfile.js]: looks like it. disregard

File: gulpfile.js (hunk 3 above)
Comment (NinjaManatee) [reply gulpfile.js]: I think so. If you close finsemble in dev mode and OF closes, it kills the gulpfile execution.

**Pull request merged**